etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/RuleEngine.java

```diff
@@ -0,0 +1,60 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine;
+
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+
+/** Manages the application of rules loaded from a definitions file using the RuleLoader. */
+public class RuleEngine {
+
+    private final Path RULES_DEFINITIONS_PATH =
+            Path.of("../etor/src/main/resources/rule_definitions.json");
+    private boolean rulesLoaded = false;
+
+    private static final RuleEngine INSTANCE = new RuleEngine();
+
+    final List<Rule> rules = new ArrayList<>();
```

Comment:
I thought having a list of rules instead of the map will give more flexibility and a consistent self-contained structure. It makes it easier to dynamically add/remove rules, which I think it will happen more often that accessing a specific rule by name. It will also preserve the order which may be important in the future

Reply:
thoughts on using a synchronized block for writing or reading to the list? I think there is a synchronized list and a CopyOnWriteArrayList that can provide thread safety. We would have to see how it will impact the performance in order to use them as options. But if the set of rules will not change frequently, thoughts on using an immutable collection to not get hit on performance as much?

Reply:
That's a good point. I don't expect the json file will be updated frequently so I don't think that would be a concern. I'm not very familiar with immutable collections in Java so I'd need to look into that